catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java

```diff
@@ -156,9 +156,17 @@ public MetacardImpl(Metacard metacard, MetacardType type) {
             throw new IllegalArgumentException(
                     MetacardType.class.getName() + " instance should not be null.");
         }
+
+        if (metacard.getSourceId() != null) {
+            setSourceId(metacard.getSourceId());
+        }
+
         map = new HashMap<>();
-        for (AttributeDescriptor attribute : metacard.getMetacardType().getAttributeDescriptors()) {
-            map.put(attribute.getName(), metacard.getAttribute(attribute.getName()));
+        for (AttributeDescriptor attribute : metacard.getMetacardType()
```

Comment:
Why does this constructor try to iterate over the attributes instead of using `wrappedMetacard` like the other constructor?  Seems like that was inconsistent before this PR.